collect/cache/span_cache_basic.go

```diff
@@ -0,0 +1,127 @@
+package cache
+
+import (
+	"sort"
+	"sync"
+
+	"github.com/facebookgo/startstop"
+	"github.com/honeycombio/refinery/types"
+	"github.com/jonboulle/clockwork"
+)
+
+// this is a naive implementation that uses a map
+// and doesn't try to be clever about memory usage or allocations
+type spanCache_basic struct {
+	Clock   clockwork.Clock `inject:""`
+	cache   map[string]*types.Trace
+	current []string
```

Comment:
Can we add a comment for `current` and `nettix` clarifying that they are not protected by the mutex?